haystack/components/fetchers/link_content.py

```diff
@@ -94,10 +95,11 @@ def __init__(
 
         # register default content handlers that extract data from the response
         self.handlers: Dict[str, Callable[[Response], ByteStream]] = defaultdict(lambda: _text_content_handler)
-        self.handlers["text/html"] = _text_content_handler
-        self.handlers["text/plain"] = _text_content_handler
-        self.handlers["application/pdf"] = _binary_content_handler
-        self.handlers["application/octet-stream"] = _binary_content_handler
+        self.handlers["text/*"] = _text_content_handler
+        self.handlers["application/json"] = _text_content_handler
+        self.handlers["application/*"] = _binary_content_handler
+        self.handlers["audio/*"] = _binary_content_handler
+        self.handlers["video/*"] = _binary_content_handler
```

Comment:
Content-Type may also be an `image` or `multipart`. Any particular reason they are not assigned handlers?

Reply:
Good point. I thought that single link can not contain multipart but image are definitely missing. Will add them and double check the multipart 🙏